src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -291,7 +291,9 @@ const connectInfiniteHits: InfiniteHitsConnector = function connectInfiniteHits(
         }
 
         const isFirstPage = getFirstReceivedPage() === 0;
-        const isLastPage = results.nbPages <= results.page + 1;
+        const isLastPage =
+          results.nbPages <= results.page + 1 ||
```

Comment:
is this case still needed, does `getLastReceivedPage` not cover the case?

Reply:
Ah, that's true. The current page is always equal or lower than the last received page.

Reply:
That's right — a badly written test was failing and got me to keep that line.

Addressed in 1e76cb1e23f6d8153effd98a5f70b77f96f8deea.